core/vm/instructions.go

```diff
@@ -829,6 +829,22 @@ func opSelfdestruct(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext
 	return nil, errStopToken
 }
 
+func opSelfdestruct6780(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]byte, error) {
+	if interpreter.readOnly {
+		return nil, ErrWriteProtection
+	}
+	beneficiary := scope.Stack.pop()
+	balance := interpreter.evm.StateDB.GetBalance(scope.Contract.Address())
+	interpreter.evm.StateDB.SubBalance(scope.Contract.Address(), balance)
+	interpreter.evm.StateDB.AddBalance(beneficiary.Bytes20(), balance)
+	interpreter.evm.StateDB.SendAll(scope.Contract.Address())
```

Comment:
I think the semantics are a bit wonky here, making it so that things are un-untuitive. 

First of all, you explicitly send all balance from source to benefiary. Fine. Then you invoke `SendAll`. But `SendAll` does _not_ send all (you already did that), all it does is set a boolean and journal that event. 

So IMO the statedb `SendAll` should be renamed into `SetSendalledFlag`, or, maybe something better. But we should remove the connotation of "this method sends all the balance", since that's the one thing that does not happen here.

Reply:
I've removed references to "sendall" because it doesn't appear in EIP-6780 (only 4758).  I'm now referring to it as `Selfdestruct6780` for lack of a better name.